packages/commonwealth/client/scripts/views/pages/notification_settings/index.tsx

```diff
@@ -56,6 +60,22 @@ const NotificationSettingsPage = () => {
     relevantSubscribedCommunities,
   } = useNotificationSettings();
 
+  const { mutateAsync: registerClientRegistrationToken } =
+    trpc.subscription.registerClientRegistrationToken.useMutation();
+
+  const handlePushNotificationSubscription = useCallback(() => {
+    async function doAsync() {
+      const permission = await Notification.requestPermission();
+      if (permission === 'granted') {
+        console.log('Notification permission granted.');
+        const token = await getFirebaseMessagingToken();
+        await registerClientRegistrationToken({ id: 'none', token });
+      }
```

Comment:
what happens if the permission is not granted?

Reply:
Nothing.  We don't send the registration token and nothing happens.